owa/app/js/components/manage-autogeneration-option/ManageAutogenerationOption.jsx

```diff
@@ -0,0 +1,455 @@
+        autogenerateSource: '',
+        autoGenerationEnabled: false,
+        manualEntryEnabled: false
+      },
+      saveGenOptionStatus: 'new',
+      errors: {},
+      alertVisible: false
+    };
+    this.getGenDetails = this.getGenDetails.bind(this);
+    this.toggle = this.toggle.bind(this);
+    this.createGenOptions = this.createGenOptions.bind(this);
+    this.deleteGenOption = this.deleteGenOption.bind(this);
+    this.nextStage = this.nextStage.bind(this);
+    this.onDismiss = this.onDismiss.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.saveNewAutoGenOptions = this.saveNewAutoGenOptions.bind(this);
+    this.editGenOption = this.editGenOption.bind(this);
+    this.editCurrentGenOption = this.editCurrentGenOption.bind(this);
+  }
+
+  getGenDetails(id) {
+    const selectedGenOption = this.state.genOptions[id];
+    this.setState({
+      saveGenOptionStatus: 'edit',
+      firstLevelCompleted: true,
+      modal: true,
+      currentGenOption: {
+        id,
+        identifierType: selectedGenOption.idType,
+        currentLocation: selectedGenOption.location,
+        autogenerateSource: selectedGenOption.sourceName,
+        autoGenerationEnabled: selectedGenOption.autoGen,
+        manualEntryEnabled: selectedGenOption.manualEntry
+      },
+    });
+  }
+
+  editCurrentGenOption() {
+    const { genOptions, currentGenOption } = this.state;
+    const { id, identifierType, currentLocation, autogenerateSource,
+        autoGenerationEnabled, manualEntryEnabled } = currentGenOption;
+    const selectedGenOption = genOptions[id];
+    selectedGenOption.idType = identifierType;
+    selectedGenOption.location = currentLocation;
+    selectedGenOption.sourceName = autogenerateSource;
+    selectedGenOption.autoGen = autoGenerationEnabled;
+    selectedGenOption.manualEntry = manualEntryEnabled;
+    genOptions[id] = selectedGenOption; 
+    this.setState({
+      genOptions,
+      modal: false
+    });
+  }
+
+  onChange(e) {
+    let currentGenOption;
+    if (e.target.name === 'autoGenerationEnabled' || e.target.name === 'manualEntryEnabled') {
+      currentGenOption = Object.assign({}, this.state.currentGenOption, { [e.target.name]: e.target.checked});
+    } else {
+      currentGenOption = Object.assign({}, this.state.currentGenOption, { [e.target.name]: e.target.value});
+    }
+    this.setState({
+      alertVisible: false,
+      currentGenOption
+    })
+  }
+
+  deleteGenOption(id) {
+    console.log('delete', id);
+  }
+
+  createGenOptions() {
+    this.setState({
+      saveGenOptionStatus: 'new',
+      modal: true,
+      firstLevelCompleted: false,
+      currentGenOption: {
+        identifierType: '',
+        currentLocation: '',
+        autogenerateSource: '',
+        autoGenerationEnabled: false,
+        manualEntryEnabled: false
+      }
+    });
+  }
+
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
+    });
+  }
+
+  editGenOption(id) {
```

Comment:
Why is this method empty? You should remove it if you no longer need it.

Reply:
I might still have need of this for the real implementation.